phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -2957,6 +3006,16 @@ MutationState dropTable(String schemaName, String tableName, String parentTableN
                 Delete linkDelete = new Delete(linkKey, clientTimeStamp);
                 tableMetaData.add(linkDelete);
             }
+            if (tableType == PTableType.TABLE) {
+                // acquire a mutex on the table to prevent creating views while concurrently
+                // dropping the base table
+                acquiredMutex = writeCell(null, schemaName, tableName, null);
+                if (!acquiredMutex) {
+                    logger.info("Failed to acquire mutex on physical table " + physicalTableName);
```

Comment:
nit: You can push mutex related logs lines to the specific methods.

Reply:
Done, can I get a +1?